src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -105,23 +110,34 @@ public function add(array $headers)
      * @param string|null $default The default value
      * @param bool        $first   Whether to return the first value or all header values
      *
-     * @return string|string[]|null The first header value or default value if $first is true, an array of values otherwise
+     * @return string|string[]|int|null The first header value or default value if $first is true, an array of values otherwise
      */
-    public function get($key, $default = null, $first = true)
+    public function get($key, $default = null)
     {
-        $key = str_replace('_', '-', strtolower($key));
-        $headers = $this->all();
+        $all = $this->getValues($key);
 
-        if (!\array_key_exists($key, $headers)) {
-            if (null === $default) {
-                return $first ? null : [];
-            }
+        if (3 === \count(\func_get_args()) && false === func_get_arg(2)) {
+            @trigger_error(sprintf('Passing $first as a third argument is deprecated since Symfony 4.4 and will be removed in Symfony 5.0. %s will return the first header value, use %s::getValues() instead to get an array of values.', __METHOD__, __CLASS__), E_USER_DEPRECATED);
 
-            return $first ? $default : [$default];
+            return $all;
         }
+        if (\count($all)) {
+            return isset($all[0]) ? $all[0] : $default;
+        }
+
+        return $default;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getValues(string $key): array
```

Comment:
Should be named `getAll()` as suggested before: this would keep consistency with similar places.

Reply:
Hmm, actually the common name is `all()` (there's one `getAll()` in Mime which should be renamed actually)

Reply:
`all` would mean to me that it returns all headers

Reply:
I agree with @Tobion and the coding standards do tell the same. Also, that that method already exist with this exact semantic.